driver/src/main/java/org/neo4j/driver/v1/Config.java

```diff
@@ -311,47 +327,54 @@ public ConfigBuilder withLeakedSessionsLogging()
          * The max number of sessions to keep open at once. Configure this
          * higher if you want more concurrent sessions, or lower if you want
          * to lower the pressure on the database instance.
-         *
+         * <p>
          * If the driver is asked to provide more sessions than this, it will
          * block waiting for another session to be closed, with a timeout.
+         * <p>
+         * Method is deprecated and will forward the given argument to {@link #withMaxConnectionPoolSize(int)}.
          *
          * @param size the max number of sessions to keep open
          * @return this builder
+         * @deprecated please use {@link #withMaxConnectionPoolSize(int)} instead.
          */
         @Deprecated
         public ConfigBuilder withMaxSessions( int size )
         {
-            return this;
+            return withMaxConnectionPoolSize( size );
         }
 
         /**
          * The max number of idle sessions to keep open at once. Configure this
          * higher if you want more concurrent sessions, or lower if you want
          * to lower the pressure on the database instance.
+         * <p>
+         * Method is deprecated and will forward the given argument to {@link #withMaxConnectionPoolSize(int)}.
          *
          * @param size the max number of idle sessions to keep open
          * @return this builder
-         * @deprecated please use {@link #withMaxIdleConnections(int)} instead.
+         * @deprecated please use {@link #withMaxConnectionPoolSize(int)} instead.
          */
         @Deprecated
         public ConfigBuilder withMaxIdleSessions( int size )
         {
-            this.maxIdleConnectionPoolSize = size;
-            return this;
+            return withMaxConnectionPoolSize( size );
```

Comment:
I was about to suggest to give a warning here, but then realized we might do not have a logger till this moment...

I feel we should not delegate maxIdle to max, as this config changes meaning already. 
I am wondering if it makes sense to just ignore the value we set to this method. What do you think of the following changes?

```
MaxPoolSize -> MaxConnectionPoolSize
MaxIdleConnections -> ignored
MaxIdleSessions -> ignored
```

Reply:
Since we do not have any idle connection checking in place on the async implementation, ignoring `MaxIdleConnections` and `MaxIdleSessions` config values makes more sense for me.

Reply:
Thanks for pointing this out. I agree, will change